jgiven-spock/build.gradle

```diff
@@ -11,7 +11,7 @@ dependencies {
     implementation 'org.codehaus.groovy:groovy:3.0.7'
     implementation 'org.spockframework:spock-core:2.0-M4-groovy-3.0'
     implementation "org.spockframework:spock-junit4:2.0-M5-groovy-3.0"
-    testCompile project(':jgiven-html5-report')
+    testImplementation project(':jgiven-html5-report')
 }
 
 test.finalizedBy(jgivenHtml5Report)
```

Comment:
this looks broken from a previous update of libraries.
If we use spock2, then we need to add the JUnit platform

```
test {
    useJUnitPlatform()
}
```

I guess that tests are not running, nor failing

Reply:
excellent point.  I wanted to revert that change anyway and have a dedicated spock2 package...
Btw. since I assume that there still will be several pushes to this branch, before this PR is actually ready to be reviewed, 
should I just inform you once it is ready? Your input is appreciated, but I'm not sure if it makes sense to review this half ready PR.

Reply:
I think it would be great to have 2 different packages, one for spock1 and another one for spock2.
I had this draft https://github.com/TNG/JGiven/pull/445 waiting until Spock 2 makes a final version, but ideally, spock2 will be using JUnit5 approach instead of JUnit4

No worries, sometimes I have some spare time in the mornings 🤗

Reply:
I checked the spock plugin this morning, and your concerns were appropriate. There are indeed failing tests in the spock plugin. 
Unfortunately, just reverting back to spock 1 is probably not an option at this point, because gradle7 uses groovy 3. Using that and groovy-2.x together didn't work in the gradle-plugin. 

My current plan is to make a separate branch (to be merged before this one) in which the spock package is built correctly. (preferaably with spock2) and then adapt this PR accordingly.

Reply:
I think we should support Spock1 and Spock2, having 2 different packages. 
Spock2 has not a release candidate yet, and we should not enforce upgrading it.